docs/design/fabric-v2+/attestation-interface.md

```diff
@@ -0,0 +1,63 @@
+# Attestation Interface
+
+This document defines the interface for producing and verifying an attestation.
+
+
+## Producing an attestation
+
+This interface is only available inside an enclave.
+
+```
+int init_attestation(
+    uint8_t* params,
+    uint32_t params_length);
+
+```
+```
+int get_attestation(
+    uint8_t statement[32],
+    uint8_t* attestation,
+    uint32_t attestation_length);
```

Comment:
attestation is what we call attestation_evidence in the uml diagrams right? just to clarify

Reply:
In theory, yes.
In practice, it could be the quote to be converted in a verification report (which is the attestation evidence).

Reply:
hmm, given that this API is an abstraction and, from discussion around `mrenclave_t`, you also aim to be independent not only of EPID and DCAP but even of SGX itself, then for this to work you will need an additional API function which does the conversion?  Otherwise you completely broke the abstraction? Or put differently(i.e., as-is), everybody looking at the public API and seeing a generation and a verification function with matching parameter names and no other function will assume to be output of `get_attestation` to be the input to `verify_attestation`.  This interpretation would also be re-enforced by looking a the current UMLs where there are evidence bytes passed from fpc stub to ercc with no documented manipulation?